### PR TITLE
Update and modernize the flutter_markdown package

### LINKED ISSUE #2591: MarkdownStyleRaw optimisations
Right now you can't create MarkdownStyleRaws on the fly (since that will force a reparse each time, since they have no operator==), and you can't create constants for them (since they don't have const constructors). We can't just give them operator==s, because then they'd do redundant creations of the map each time. Maybe we can have the map be provided by a function that the consumer calls once, and then turn the rest of the class into a normal const constructable object with operator==?

### LINKED ISSUE #3123: markdown library dartdocs are copypasta'ed to multiple classes
The docs for Markdown, MarkdownBody, MarkdownBodyRaw, and MarkdownRaw are all the same. Also MarkdownStyle and MarkdownStyleRaw.

### LINKED ISSUE #6166: flutter_markdown isn't maintained
As we're stabilizing our APIs, we might want to take a look at https://github.com/flutter/flutter/tree/master/packages/flutter_markdown For example, not sure if we're testing it.

Let's decide what we'd like to do with this library.

### PR DESCRIPTION
**Pull request by abarth**
We now use modern scrolling machinery and patterns. The API should also be
easier to maintain over time.

Fixes #6166
Fixes #2591
Fixes #3123

### CODE REVIEW

**Comment by abarth** [comment]
@Hixie

File: packages/flutter_markdown/lib/src/builder.dart [hunk 1]
```diff
@@ -0,0 +1,243 @@
...
+  'blockquote',
+  'img',
+  'pre',
+  'ol',
+  'ul',
+];
+
+const List<String> _kListTags = const <String>['ul', 'ol'];
+
+bool _isBlockTag(String tag) => _kBlockTags.contains(tag);
```
Comment (Hixie) [inline packages/flutter_markdown/lib/src/builder.dart]: that's expensive. pity that _kBlockTags can't be a const Set.

Comment (Hixie) [inline packages/flutter_markdown/lib/src/builder.dart]: https://github.com/dart-lang/sdk/issues/174

File: packages/flutter_markdown/lib/src/builder.dart [hunk 2]
```diff
@@ -0,0 +1,243 @@
...
+  final List<TextSpan> children = <TextSpan>[];
+}
+
+/// A delegate used by [MarkdownBuilder] to control the widgets it creates.
+abstract class MarkdownBuilderDelegate {
+  /// Returns a gesture recognizer to use for an `a` element with the given
+  /// `href` attribute.
+  GestureRecognizer createLink(String href);
+
+  /// Returns formated text to use to display the given contents of a `pre`
```
Comment (Hixie) [inline packages/flutter_markdown/lib/src/builder.dart]: formatted

File: packages/flutter_markdown/lib/src/style_sheet.dart [hunk 3]
```diff
@@ -0,0 +1,269 @@
...
+
+  /// A [Map] from element name to the cooresponding [TextStyle] object.
+  Map<String, TextStyle> get styles => _styles;
+  Map<String, TextStyle> _styles;
+
+  @override
+  bool operator ==(dynamic other) {
+    if (identical(this, other))
+      return true;
+    if (other is! MarkdownStyleSheet)
```
Comment (Hixie) [inline packages/flutter_markdown/lib/src/style_sheet.dart]: compare runtimeType rather than using `is`

Comment (Hixie) [inline packages/flutter_markdown/lib/src/style_sheet.dart]: (for symmetry)

File: packages/flutter_markdown/lib/src/widget.dart [hunk 4]
```diff
@@ -0,0 +1,211 @@
...
+    styleSheet: styleSheet,
+    syntaxHighlighter: syntaxHighlighter,
+    onTapLink: onTapLink,
+  );
+
+  @override
+  Widget build(BuildContext context, List<Widget> children) {
+    if (children.length == 1)
+      return children.single;
+    return new Column(
```
Comment (Hixie) [inline packages/flutter_markdown/lib/src/widget.dart]: use BlockBody instead?

**Comment by Hixie** [comment]
LGTM

File: packages/flutter_markdown/lib/src/builder.dart (hunk 1 above)
Comment (abarth) [inline packages/flutter_markdown/lib/src/builder.dart]: I can make it a final set, which is almost as good.

File: packages/flutter_markdown/lib/src/widget.dart (hunk 4 above)
Comment (abarth) [inline packages/flutter_markdown/lib/src/widget.dart]: I had BlockBody, but that failed a test.  The old code did it this way and I suspect that's what folks expect.

Comment (Hixie) [inline packages/flutter_markdown/lib/src/widget.dart]: What's the difference?

**Pull request merged**